src/xserver_zmq.cpp

```diff
@@ -6,37 +6,35 @@
 * The full license is in the file LICENSE, distributed with this software. *
 ****************************************************************************/
 
-#include "xeus/xserver_zmq.hpp"
 #include <thread>
 #include <chrono>
+
+#include "xeus/xserver_zmq.hpp"
+#include "xeus/xguid.hpp"
 #include "zmq_addon.hpp"
 #include "xmiddleware.hpp"
 #include "xpublisher.hpp"
 #include "xheartbeat.hpp"
 
 namespace xeus
 {
-    void init_socket(zmq::socket_t& socket, const std::string& end_point)
-    {
-        socket.setsockopt(ZMQ_LINGER, get_socket_linger());
-        socket.bind(end_point);
-    }
 
-    xserver_zmq::xserver_zmq(zmq::context_t& context, const xconfiguration& c)
-        : m_shell(context, zmq::socket_type::router)
-        , m_controller(context, zmq::socket_type::router)
-        , m_stdin(context, zmq::socket_type::router)
-        , m_publisher_pub(context, zmq::socket_type::pub)
-        , m_publisher_controller(context, zmq::socket_type::req)
-        , m_heartbeat_controller(context, zmq::socket_type::req)
-        , p_publisher(new xpublisher(context, c.m_transport, c.m_ip, c.m_iopub_port))
-        , p_heartbeat(new xheartbeat(context, c.m_transport, c.m_ip, c.m_hb_port))
-        , m_request_stop(false)
+    xserver_zmq::xserver_zmq(zmq::context_t& context, xconfiguration& config)
```

Comment:
The `configuration` parameter should remain constant. The implementation of the constructor is correct, do not forget to add and implement the public API for retrieving the port numbers.